### PR TITLE
fix: expand sqlite futures arbitrage schema

### PR DESCRIPTION
**Pull request by irfndi**
## Summary
- Add SQLite migration `081_expand_futures_arbitrage_opportunities.sql` to bring `futures_arbitrage_opportunities` up to the runtime insert contract.
- Add the unique `(symbol, long_exchange, short_exchange)` index required by the service upsert.
- Add a SQLite regression test that applies the old arbitrage table migration plus the new expansion migration, stores an opportunity, and verifies the upsert does not duplicate rows.

## Validation
- `rtk env TMPDIR=/private/tmp/nt-go-tmp-begin-paper GOTMPDIR=/private/tmp/nt-go-tmp-begin-paper GOCACHE=/private/tmp/nt-go-cache-begin-paper go -C services/backend-api test ./internal/services -run 'TestFuturesArbitrageService_storeOpportunity_SQLiteMigrationSchema' -count=1`
- `rtk env TMPDIR=/private/tmp/nt-go-tmp-begin-paper GOTMPDIR=/private/tmp/nt-go-tmp-begin-paper GOCACHE=/private/tmp/nt-go-cache-begin-paper go -C services/backend-api test ./internal/services -count=1`
- `rtk env TMPDIR=/private/tmp/nt-go-tmp-begin-paper GOTMPDIR=/private/tmp/nt-go-tmp-begin-paper GOCACHE=/private/tmp/nt-go-cache-begin-paper make build`

Fixes bd `NeuraTrade-e15`, discovered from `neura-zuju`.

## Summary by Sourcery

Align SQLite futures arbitrage opportunities schema with the service’s runtime insert contract and ensure upserts remain unique per opportunity.

Bug Fixes:
- Prevent duplicate SQLite futures arbitrage opportunity rows when upserting the same symbol and exchange pair.

Enhancements:
- Expand the SQLite futures_arbitrage_opportunities table with additional fields and indexes to match the richer Postgres-oriented model shape.

Tests:
- Add a regression test that applies legacy and new SQLite migrations and verifies futures arbitrage upserts do not create duplicate records.

## Summary by CodeRabbit

* **Chores**
  * Enhanced database infrastructure for futures arbitrage opportunities with expanded metrics including pricing, funding, risk, liquidity, and leverage data.
  * Strengthened test coverage for arbitrage opportunity data handling and schema validation.

### CODE REVIEW

File: services/backend-api/internal/services/futures_arbitrage_service_test.go [hunk 1]
```diff
@@ -302,6 +304,115 @@ func TestFuturesArbitrageService_storeOpportunity(t *testing.T) {
...
+	err = sqliteDB.QueryRow(
+		context.Background(),
+		`SELECT COUNT(*) FROM futures_arbitrage_opportunities
+		 WHERE symbol = $1 AND long_exchange = $2 AND short_exchange = $3`,
+		"BTC/USDT",
+		"binance",
+		"bybit",
+	).Scan(&count)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, count, "unique upsert should update the existing opportunity")
```
[the remark this answers is not in the document]
Comment (irfndi) [reply services/backend-api/internal/services/futures_arbitrage_service_test.go:423]: Addressed in `42eda0d`: the regression now reads `apy` after the second `storeOpportunity` call and asserts it was updated to `0.42`, in addition to the single-row upsert count check.

**Comment by irfndi** [comment]
Review follow-up: the current head `42eda0d` addresses the Sourcery high-level notes by moving SQLite migration application into the shared `applySQLiteMigrationBySuffix` test helper and discovering migration files by logical suffix instead of hard-coding full numbered filenames in the regression test. The same commit also extends the upsert regression to assert the APY value is updated in place.
